ride_share.rb

```diff
@@ -0,0 +1,146 @@
+rides = [
+  {
+    driver_id: 'DR0004',
+    date: '3rd Feb 2016',
+    cost: '5',
+    rating: '5',
+    rider_id: 'RD0022', 
+  },
+  {
+    driver_id: 'DR0001',
+    date: '3rd Feb 2016',
+    cost: '10',
+    rating: '3',
+    rider_id: 'RD0003',
+  },
+  {
+    driver_id: 'DR0002',
+    date: '3rd Feb 2016',
+    cost: '25',
+    rating: '5',
+    rider_id: 'RD0073'
+  },
+  {
+    driver_id: 'DR0001',
+    date: '3rd Feb 2016',
+    cost: '30',
+    rating: '4',
+    rider_id: 'RD0015'
+  },
+  {
+    driver_id: 'DR0003',
+    date: '4th Feb 2016',
+    cost: '5',
+    rating: '5',
+    rider_id: 'RD0066'
+  },
+  {
+    driver_id: 'DR0004',
+    date: '4th Feb 2016',
+    cost: '10',
+    rating: '4',
+    rider_id: 'RD0022'
+  },
+  {
+    driver_id: 'DR0002',
+    date: '4th Feb 2016',
+    cost: '15',
+    rating: '1',
+    rider_id: 'RD0013'
+  },
+  {
+    driver_id: 'DR0003',
+    date: '5th Feb 2016',
+    cost: '50',
+    rating: '2',
+    rider_id: 'RD0003'
+  },
+  {
+    driver_id: 'DR0002',
+    date: '5th Feb 2016',
+    cost: '35',
+    rating: '3',
+    rider_id: 'RD0066'
+  },
+  {
+    driver_id: 'DR0004',
+    date: '5th Feb 2016',
+    cost: '20',
+    rating: '5',
+    rider_id: 'RD0073'
+  },
+  {
+    driver_id: 'DR0001',
+    date: '5th Feb 2016',
+    cost: '45',
+    rating: '2',
+    rider_id: 'RD0003'
+  }
+]
+
+driver_ids = ["DR0001", "DR0002", "DR0003", "DR0004"]
+
+# Question 1: The number of rides each driver has given?
+def driver_ride_count(driver_id, rides)
+  driver_rides = rides.select { |ride|  ride [:driver_id] == driver_id  }
```

Comment:
I think it's interesting that you start this method with `select`, because it means that the first thing that you do is group all of your rides by driver. This process costs your program a little bit of time as it has to loop through the whole array. Could you remove this extra work by restructuring your data?